pkg/pbflow/proto.go

```diff
@@ -94,6 +97,18 @@ func FlowToPB(fr *flow.Record) *Record {
 		pbflowRecord.Network.SrcAddr = &IP{IpFamily: &IP_Ipv4{Ipv4: flow.IntEncodeV4(fr.Id.SrcIp)}}
 		pbflowRecord.Network.DstAddr = &IP{IpFamily: &IP_Ipv4{Ipv4: flow.IntEncodeV4(fr.Id.DstIp)}}
 	}
+	if s != nil {
+		for _, metadata := range fr.Metrics.NetworkEvents {
+			if !flow.AllZerosMetaData(metadata) {
+				if md, err := s.DecodeCookie8Bytes(metadata); err == nil {
```

Comment:
I'm not convinced that doing this decoding within the protobuf encoding is the best option we have, it doesn't have to be tied to the transport data model - also I'm not sure if the DirectFLP mode uses pb, I don't think? So that would mean to duplicate in several places the decoding part.

I don't think it's a blocker for the current work, but maybe to consider as a follow-up? For instance, we could create a new stage in `buildAndStartPipeline` (in agent.go) to do the decoding.

Reply:
I was looking at the direct-flp path and it does populate the cm directly but yeah we can follow up with further enhancement as needed that was the best I can come up with with less risk